solr/solrj/src/java/org/apache/solr/common/cloud/ZkStateReader.java

```diff
@@ -602,8 +740,13 @@ private void constructState(Set<String> changedCollections) {
     Map<String, ClusterState.CollectionRef> result = new LinkedHashMap<>(legacyCollectionStates);
 
     // Add state format2 collections, but don't override legacy collection states.
-    for (Map.Entry<String, DocCollection> entry : watchedCollectionStates.entrySet()) {
-      result.putIfAbsent(entry.getKey(), new ClusterState.CollectionRef(entry.getValue()));
+    for (Entry<String, StatefulCollectionWatch> entry :
+        collectionWatches.watchedCollectionEntries()) {
+      if (entry.getValue().currentState != null) {
+        // if the doc is null for the collection watch, then it should not be inserted into the
```

Comment:
is it some specific condition here?

Reply:
Yes. This is to maintain the consistency with the old behavior, which a null state wuold have no key in `watchedCollectionStates` https://github.com/fullstorydev/lucene-solr/pull/176/files/44b45f0d3ef418815299961f8269b283bdf9a207#diff-c797d60b04b72ceaf84395b8a77c9c51299259d4214e58bcd92331c2bb9c51acL2079 . Hence not inserted into the result